src/components/forms/CharacterCount/CharacterCount.tsx

```diff
@@ -71,17 +71,23 @@ export const CharacterCount = ({
   const [length, setLength] = useState(initialCount)
   const [message, setMessage] = useState(getMessage(initialCount, maxLength))
   const [isValid, setIsValid] = useState(initialCount < maxLength)
+  const srMessageRef = useRef<HTMLDivElement>(null)
 
   const classes = classnames('usa-character-count__field', className)
-  const messageClasses = classnames(
-    'usa-hint',
-    'usa-character-count__message',
-    { 'usa-character-count__message--invalid': !isValid }
-  )
+  const messageClasses = classnames('usa-hint', 'usa-character-count__status', {
+    'usa-character-count__status--invalid': !isValid,
+  })
 
   useEffect(() => {
-    setMessage(getMessage(length, maxLength))
+    const message = getMessage(length, maxLength)
+    setMessage(message)
     setIsValid(length <= maxLength)
+    // Updates the character count status for screen readers after a 1000ms delay
+    const timer = setTimeout(() => {
+      // Setting the text directly for VoiceOver compatibility.
+      if (srMessageRef.current) srMessageRef.current.textContent = message
+    }, 1000)
+    return () => clearTimeout(timer)
```

Comment:
**question:** Does a useState/setState not work for Voiceover?

Reply:
It does not! Very weird that it didn't. It _did_ work with every other SR I tested.